VirtoCommerce.ContentModule.Web/Scripts/blades/pages/edit-blog.js

```diff
@@ -42,29 +42,48 @@
         blade.currentEntity.dynamicProperties = props;
     }
 
-    $scope.saveChanges = function () {
-        blade.isLoading = true;
+    $scope.saveWithMetadata = function() {
 
         contentApi.saveWithMetadata({
-            contentType: blade.contentType,
-            storeId: blade.storeId,
-            folderUrl: ''
-        }, {
-            dynamicProperties: blade.currentEntity.dynamicProperties,
-            content: '',
-            name: getBlogBlobName()
-        },
-        function (data) {
-            blade.isLoading = false;
-            blade.origEntity = angular.copy(blade.currentEntity);
-            if (blade.isNew) {
-                $scope.bladeClose();
-                $rootScope.$broadcast("cms-statistics-changed", blade.storeId);
-            }
+                contentType: blade.contentType,
+                storeId: blade.storeId,
+                folderUrl: blade.currentEntity.name
+            },
+            {
+                dynamicProperties: blade.currentEntity.dynamicProperties,
+                content: '',
+                name: getBlogBlobName()
+            },
+            function (data) {
+                blade.isLoading = false;
+                blade.origEntity = angular.copy(blade.currentEntity);
+                if (blade.isNew) {
+                    $scope.bladeClose();
+                    $rootScope.$broadcast("cms-statistics-changed", blade.storeId);
+                }
 
-            blade.parentBlade.refresh();
-        },
-        function (error) { bladeNavigationService.setError('Error ' + error.status, $scope.blade); blade.isLoading = false; });
+                blade.parentBlade.refresh();
+            },
+            function (error) {
+                bladeNavigationService.setError('Error ' + error.status, $scope.blade);
+                blade.isLoading = false;
+            });
+    };
+
+    $scope.saveChanges = function () {
+        blade.isLoading = true;
+
+        var parentUrl = blade.currentEntity.url;
+        var folderName = blade.currentEntity.name;
+        if (!parentUrl) {
+            contentApi.createFolder(
+                { contentType: blade.contentType, storeId: blade.storeId },
+                { name: folderName, parentUrl: parentUrl },
+                $scope.saveWithMetadata(),
+                function(error) { bladeNavigationService.setError('Error ' + error.status, blade); });
```

Comment:
I think we do not need error handler here, as it should be handled by common httpInterceptor. Could you check?